src/main/java/org/hiero/sketch/dataset/LocalDataSet.java

```diff
@@ -0,0 +1,47 @@
+package org.hiero.sketch.dataset;
+
+import org.hiero.sketch.dataset.api.*;
+import rx.Observable;
+
+public class LocalDataSet<T> implements IDataSet<T> {
+    private final T data;
+    // TODO: run on a separate thread
+
+    public LocalDataSet(final T data) {
+        this.data = data;
+    }
+
+    @Override
+    public <S> Observable<PartialResult<IDataSet<S>>> map(final IMap<T, S> mapper) {
+        final PartialResult<S> initial = new PartialResult<S>(0.0, null);
+        final Observable<PartialResult<S>> start = Observable.just(initial);
+        final Observable<PartialResult<S>> trsf = mapper.apply(this.data);
```

Comment:
Any particular reason you're referring to member variables with "this." all the time?